### PR TITLE
Use global logger for xorm logs and add options

### PR DESCRIPTION
**Pull request by xoxys**
Please let me know if this goes in the right direction. Needs some tests and docs.

### CODE REVIEW

File: server/store/common.go [hunk 1]
```diff
@@ -14,8 +14,14 @@
...
+type XORM struct {
+	Log     bool
+	ShowSQL bool
+}
+
 // Opts are options for a new database connection
 type Opts struct {
 	Driver string
 	Config string
+	XORM   *XORM
```
Comment (6543) [inline server/store/common.go:26]:
```suggestion
	XORM   XORM
```

make sure it's not nil

File: server/store/datastore/xorm.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+package datastore
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	xlog "xorm.io/xorm/log"
+)
+
+func NewXORMLogger(level xlog.LogLevel) *XORMLogger {
```
Comment (6543) [inline server/store/datastore/xorm.go:24]: unexport and return type can be xorm-logger-interface directly

File: server/store/datastore/xorm.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	return &XORMLogger{
+		level: level,
+	}
+}
+
+// XORMLogger custom log implementation for ILogger
+type XORMLogger struct {
```
Comment (6543) [inline server/store/datastore/xorm.go:31]: unexporte

**Comment by 6543** [comment]
yes that's the right direction :+1:  !

File: server/store/datastore/xorm.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+// xormLogger custom log implementation for ILogger
+type xormLogger struct {
+	level   xlog.LogLevel
+	showSQL bool
+}
+
+// Error implement ILogger
+func (x *xormLogger) Error(v ...interface{}) {
+	if x.level <= xlog.LOG_ERR {
+		log.Error().Msg(fmt.Sprintln(v...))
```
Comment (6543) [inline server/store/datastore/xorm.go:39]: we should add some string to indicate it's from xorm

log.Str("component", "xorm")

or log.Bool("xorm": true)
....

?

Comment (6543) [reply server/store/datastore/xorm.go:39]: what do you think?

Comment (xoxys) [reply server/store/datastore/xorm.go:39]: Yeah, that makes sense to me :+1:  Will finish the PR later.

Comment (xoxys) [reply server/store/datastore/xorm.go]: I have used the `"component", "xorm"` format. If someone wants to parse the logs in a central log mgm system, using a consistent field to get the component name might be better than a changing bool field.

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
No, just change current docs ...

And on branch out, versionated docs are generated ...

**Pull request merged**